packages/financial-templates-lib/src/price-feed/EthVixPriceFeed.js

```diff
@@ -0,0 +1,114 @@
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+
+    this.uuid = `dVIX.${inverse ? "iethVIX" : "ethVIX"}`;
+    this.historicalPrices = [];
+  }
+
+  getCurrentPrice() {
+    assert(this.lastUpdateTime, `${this.uuid}: undefined lastUpdateTime. Update required.`);
+    return this.currentPrice;
+  }
+
+  async getHistoricalPrice(time) {
+    assert(this.lastUpdateTime, `${this.uuid}: undefined lastUpdateTime. Update required.`);
+
+    assert(
+      moment.utc(time).isAfter(this.historicalPrices[0].timestamp),
+      `${this.uuid}: The requested time precedes available data.`
+    );
+
+    // Rounds timestamp down to the nearest 15m, the minimum index update frequency
+    let roundedTime = moment.utc(time).startOf("minute");
+    if (roundedTime.minutes() % 15) {
+      roundedTime = roundedTime.subtract(roundedTime.minutes() % 15, "minutes");
+    }
+
+    const result = this.historicalPrices.find(price => roundedTime.isSame(price.timestamp));
+    assert(result, `${this.uuid}: No cached result found for timestamp: ${roundedTime.toISOString()}`);
+
+    return parseFixed(result.vix, 6);
+  }
+
+  getLastUpdateTime() {
+    return this.lastUpdateTime;
+  }
+
+  async update() {
+    const currentTime = this.getTime();
+    const earliestAllowableUpdateTime = currentTime + moment.duration(this.minTimeBetweenUpdates, "seconds");
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== undefined && moment(currentTime).isSameOrAfter(earliestAllowableUpdateTime)) {
+      console.log({
+        at: "ETHVIXPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: earliestAllowableUpdateTime - currentTime
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "ETHVIXPriceFeed",
+      message: "Updating",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime
+    });
+
+    // 1. Request the data.
+    const priceUrl = "https://dvix.io/api/historicalData?currency=ETH";
+    const response = await this.networker.getJson(priceUrl);
+
+    // 2. Check the response.
+    assert(
+      Array.isArray(response) && response.length,
+      `🚨 Could not fetch historical prices from url ${priceUrl}: ${JSON.stringify(response)}`
+    );
+
+    // Expected response data structure:
+    // [
+    //   {
+    //     "timestamp": "2021-03-24T15:00:00.000Z",
+    //     "iVix": 142.4470728085134,
+    //     "vix": 70.20151276427174,
+    //     ...
+    //   },
+    //   ...
+    // ]
+
+    // 3. Get last result in the stack.
+    const mostRecent = response[response.length - 1];
+
+    // 4. Store results.
+    this.lastUpdateTime = currentTime;
+    this.historicalPrices = [...this.historicalPrices, ...response];
+    this.currentPrice = parseFixed(this.inverse ? mostRecent.iVix : mostRecent.vix, 6);
```

Comment:
Here, you are scaling the price to 6 decimal points. Your UMIP defines the collateral decimals as 6 and this might be the reason why you are scaling this to 6. However, I think you should be scaling your price to _18_ decimals to be consistent with what the DVM would return from a price request (the price feed should mimic what the DVM returns).

The general design of price feeds and their associated scaling is to define the `priceFeedDecimals` within the price feed to match what the DVM would return in a price request. If a voter was voting on an ethVIX request the number returned would be scaled by 1e18.

To add a bit of context, financial contracts deploy a synthetic token with synthetic decimals matching the collateral's decimals. All price feeds are scaled by 1e18. For example, a synthetic deployed with USDC collateral would have 6 decimal places to match the collateral. This example synthetic would have a price scaled by 1e18.

The only time you should be scaling a price feed to something other than 18  decimals is when you are using a legacy UMA financial contract. Previously we would scale the price request to match the collateral, leaving the synthetic at 18 decimals.

Reply:
See here: https://github.com/UMAprotocol/protocol/blob/0eea3fbb610f74694c22ca36f6902faf3fa9092b/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js#L60

for an example on how we do this in other feeds.

Reply:
This is understandable. Tbh it was a bit confusing in the docs why the precision was expected at 18 decimals, but it seems reasonable given this context. I'll make the changes shortly and push the updated branch for review.